### PR TITLE
Improve error handling when creating a new History with the master API key

### LINKED ISSUE #10178: Uncaught exception in exposed API method when trying to create new history using master api key
Hi, I'm trying to automate the provisioning of some galaxy containers using bioblend, and for that I need to be able to create new histories in which the various datasets belong. However every time I try to create a new history using the bioblend method ```instance.histories.create_history(name=xxxx)```, it causes the following error: 

```bioblend.ConnectionError: Unexpected HTTP status code: 500: {"err_msg": "Uncaught exception in exposed API method:", "err_code": 0}```

Here is the complete error log:
```
Traceback (most recent call last):
  File "/xxx/xxx/xxx/xxx/xxx/xxx/xxx/xxx/autoload.py", line 831, in <module>
    al.init_instance()
  File "/xxx/xxx/xxx/xxx/xxx/xxx/xxx/xxx/autoload.py", line 603, in init_instance
    self.connect_to_instance()
  File "/xxx/xxx/xxx/xxx/xxx/xxx/xxx/xxx/autoload.py", line 149, in connect_to_instance
    self.instance.histories.create_history(name=str(self.full_name))
  File "/xxx/xxx/xxx/xxx/xxx/xxx/xxx/lib/python3.8/site-packages/bioblend/galaxy/histories/__init__.py", line 37, in create_history
    return self._post(payload)
  File "/xxx/xxx/xxx/xxx/xxx/xxx/xxx/lib/python3.8/site-packages/bioblend/galaxy/client.py", line 172, in _post
    return self.gi.make_post_request(url, payload=payload,
  File "/xxx/xxx/xxx/xxx/xxx/xxx/xxx/lib/python3.8/site-packages/bioblend/galaxyclient.py", line 131, in make_post_request
    raise ConnectionError("Unexpected HTTP status code: %s" % r.status_code,
bioblend.ConnectionError: Unexpected HTTP status code: 500: {"err_msg": "Uncaught exception in exposed API method:", "err_code": 0}
```
I'm running bioblend 0.14.0, and it didn't happen when I was using version  0.13.0.

As the error message points to no specific reason for this failure, help would be greatly appreciated in order to determine what is going wrong. I though it might be because I use an environment "master" API key to access the galaxy instance of a galaxy container, but I can connect to it just fine using the url and this API key. 

Thank you in advance :smiley:

### PR DESCRIPTION
**Pull request by davelopez**
## What did you do? 
- Check if the user has a proper session when creating a new history from the API. This will return a `BadRequest` instead of an `InternalServerError` if, for example, the master API key is used to create a new history, since it has no session.
- Included an API test.

## Why did you make this change?
Fixes #10178

## How to test the changes? 
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/histories.py [hunk 1]
```diff
@@ -334,6 +334,8 @@ def create(self, trans, payload, **kwd):
         :rtype:     dict
         :returns:   element view of new history
         """
+        if not trans.galaxy_session:
+            raise exceptions.MessageException("Only users with a session can create histories.")
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/api/histories.py]: I think this could deserve its own exception class?

**Comment by davelopez** [comment]
Hmm... this breaks an insane amount of tests 😅 
I wonder if a session was required to create the history... why they used to work without a session... I'll investigate further.

File: lib/galaxy/webapps/galaxy/api/histories.py (hunk 1 above)
Comment (davelopez) [reply lib/galaxy/webapps/galaxy/api/histories.py]: I created a specific exception here https://github.com/galaxyproject/galaxy/pull/11846/commits/4460b0ce3476d6174c83b4026f5292bfa8b0eb14 (sorry for the auto-formatting)

**Comment by davelopez** [comment]
I changed the condition to specifically check for a `bootstrap admin user`. Hopefully, this won't affect the existing tests.

**Comment by jmchilton** [comment]
Thanks!

**Pull request merged**